### PR TITLE
(GH-405) Moved hard coded text strings to resource file

### PR DESCRIPTION
**Pull request by AdmiringWorm**
A few text and header attributes in the xaml files was not moved to the Resource.resx file.
this PR fixes that.
It also changes ResXFileCodeGenerator to PublicResXFileCodeGenerator so that if new strings are added to the Resource.rex file or the designer is regenerated it will make the properties public (which is needed to be able to use them in xaml).

#405

### CODE REVIEW

**Comment by AdmiringWorm** [comment]
ops, what a mistake to make.
I'll probably end up squashing everything before it's merge, but still I'll rebase the changes for know and correct my mistake.

**Comment by AdmiringWorm** [comment]
I've fixed the mistake, as well as added quite a few new resource strings from the C# Code.

File: Source/ChocolateyGui/Resources/Controls.xaml [hunk 1]
```diff
@@ -670,16 +670,16 @@
...
-        <MenuItem Header="Uninstall" Icon="{StaticResource UacIcon}"
+        <MenuItem Header="{x:Static lang:Resources.Controls_PackagesContextMenuUninstall}" Icon="{StaticResource UacIcon}"
                   Visibility="{Binding IsInstalled, Converter={StaticResource BooleanToVisibility}}"
                   Command="{commands:DataContextCommandAdapter Uninstall}" />
-        <MenuItem Header="Reinstall" Icon="{StaticResource UacIcon}"
+        <MenuItem Header="{x:Static lang:Resources.Controls_PackagesContextMenuReinstall}" Icon="{StaticResource UacIcon}"
                   Visibility="{Binding IsInstalled, Converter={StaticResource BooleanToVisibility}}"
                   Command="{commands:DataContextCommandAdapter Reinstall}" />
-        <MenuItem Header="Update" Icon="{StaticResource UacIcon}"
+        <MenuItem Header="{x:Static lang:Resources.Controls_PackagesContextMenuUpdate}" Icon="{StaticResource UacIcon}"
```
Comment (RichiCoder1) [inline Source/ChocolateyGui/Resources/Controls.xaml]: Hmm. Maybe at some point we should come up with a strategy so we don't duplicate resource strings needlessly?

File: Source/ChocolateyGui/Services/ChocolateyRemotePackageService.cs [hunk 2]
```diff
@@ -242,7 +282,9 @@ public ValueTask<bool> RequiresElevation()
         public void Dispose()
         {
             _logStream?.Dispose();
-            _wampChannel?.Close("Exiting", new GoodbyeDetails { Message = "Exiting" });
+            _wampChannel?.Close(
+                Resources.ChocolateyRemotePackageService_Exiting,
```
Comment (RichiCoder1) [inline Source/ChocolateyGui/Services/ChocolateyRemotePackageService.cs]: Actually, these need to stay hardcoded magic strings. I'll move them to a const later.

File: Source/ChocolateyGui/Services/ChocolateyRemotePackageService.cs [hunk 3]
```diff
@@ -283,7 +325,9 @@ private Task Initialize(bool requireAdmin = false)
                     _isInitialized = false;
                     _logStream.Dispose();
                     _logStream = null;
-                    _wampChannel.Close("Escalating", new GoodbyeDetails { Message = "Escalating" });
+                    _wampChannel.Close(
+                        Resources.ChocolateyRemotePackageService_Escalating,
+                        new GoodbyeDetails { Message = Resources.ChocolateyRemotePackageService_Escalating });
```
Comment (RichiCoder1) [inline Source/ChocolateyGui/Services/ChocolateyRemotePackageService.cs]: Same here. These are intentionally magical.

File: Source/ChocolateyGui/ViewModels/RemoteSourceViewModel.cs [hunk 4]
```diff
@@ -231,9 +232,9 @@ protected override void OnInitialize()
                 MessageBox.Show(
```
Comment (RichiCoder1) [inline Source/ChocolateyGui/ViewModels/RemoteSourceViewModel.cs]: Oh wow, this is old. I'll need to rip this out in a seperate PR.

File: Source/ChocolateyGui/ViewModels/SourcesViewModel.cs [hunk 5]
```diff
@@ -90,7 +90,7 @@ protected override void OnViewReady(object view)
         {
             Observable.FromEventPattern<PropertyChangedEventArgs>(this, nameof(PropertyChanged))
                 .Where(p => p.EventArgs.PropertyName == nameof(ActiveItem))
-                .Subscribe(p => DisplayName = $"Source - {ActiveItem.DisplayName}");
+                .Subscribe(p => DisplayName = $"Source - {ActiveItem?.DisplayName}");
```
Comment (RichiCoder1) [inline Source/ChocolateyGui/ViewModels/SourcesViewModel.cs]: 👍

File: Source/ChocolateyGui/Resources/Controls.xaml (hunk 1 above)
Comment (AdmiringWorm) [inline Source/ChocolateyGui/Resources/Controls.xaml]: Could perhaps seperate the strings out into their own Resource file, while just having common strings in the Resource.resx file?

File: Source/ChocolateyGui/Services/ChocolateyRemotePackageService.cs (hunk 2 above)
Comment (AdmiringWorm) [inline Source/ChocolateyGui/Services/ChocolateyRemotePackageService.cs]: The Message property as well?

File: Source/ChocolateyGui/ViewModels/SourcesViewModel.cs (hunk 5 above)
Comment (AdmiringWorm) [inline Source/ChocolateyGui/ViewModels/SourcesViewModel.cs]: I got so annoyed by getting a null exception every time I closed the program, so just had to add it.

File: Source/ChocolateyGui/Services/ChocolateyRemotePackageService.cs (hunk 2 above)
Comment (RichiCoder1) [inline Source/ChocolateyGui/Services/ChocolateyRemotePackageService.cs]: Yup. Ditto for other.

File: Source/ChocolateyGui/Resources/Controls.xaml (hunk 1 above)
Comment (RichiCoder1) [inline Source/ChocolateyGui/Resources/Controls.xaml]: Hmm. I think we ran into an issue where localization wasn't happening for other Resxs? Maybe have been a misconfiguration though. I'd be for a Resources_Common.resx or something like that.

File: Source/ChocolateyGui/ViewModels/SourcesViewModel.cs (hunk 5 above)
Comment (RichiCoder1) [inline Source/ChocolateyGui/ViewModels/SourcesViewModel.cs]: Yup. I have this fixed properly on another outstanding branch.

Comment (AdmiringWorm) [inline Source/ChocolateyGui/ViewModels/SourcesViewModel.cs]: Should probably revert this change then, or?

Comment (RichiCoder1) [inline Source/ChocolateyGui/ViewModels/SourcesViewModel.cs]: Go ahead and keep it. I'll fix it up in my other PR, as that's looking like it won't get merged for a few days yet still.

File: Source/ChocolateyGui/Resources/Controls.xaml (hunk 1 above)
Comment (AdmiringWorm) [inline Source/ChocolateyGui/Resources/Controls.xaml]: I can probably look into that, but that probably deserves its own PR though.

Comment (AdmiringWorm) [inline Source/ChocolateyGui/Resources/Controls.xaml]: I don't really see how it would add much complexity.
IMO it could rather be easier to maintain if strings are located in their own resource file. (Common.resx, PackageView.resx, etc.)

However, this may come down to preference though.

Comment (RichiCoder1) [inline Source/ChocolateyGui/Resources/Controls.xaml]: I'd be up for having a handful of resource category files (common, errors, UI, images), as I've done that previously and it's worked well. Per-component would be excessive IMO however.

**Comment by RichiCoder1** [comment]
Awesome work :)
